bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/visitors/IdentifyLatestRecordRowNumVisitor.java

```diff
@@ -0,0 +1,112 @@
+                    ASTNodes.ASTExpression expr = col.getExpression();
+                    if (expr instanceof ASTNodes.ASTAnalyticFunctionCall) {
+                        ASTNodes.ASTExpression astExpression = ((ASTNodes.ASTAnalyticFunctionCall) expr).getExpression();
+                        if (astExpression instanceof ASTNodes.ASTFunctionCall) {
+                            ImmutableList<ASTNodes.ASTIdentifier> identifiers = ((ASTNodes.ASTFunctionCall) astExpression).getFunction().getNames();
+                            for (ASTNodes.ASTIdentifier idString : identifiers) {
+                                if (idString.getIdString().equals(ROW_NUMBER_STR)) {
+                                    rowNumAlias = col.getAlias().getIdentifier().getIdString();
+                                    rowNumLineNum = idString.getParseLocationRange().start();
+                                    rownumFlag = true;
+                                    break;
+                                }
+
+                            }
+
+                        }
+
+                    }
+
+                }
+                if(!rownumFlag){
+                    // if sub-query doesn't have row number go to inner sub-query
+                    super.visit(selectQuery);
+                }
+            }
+        }
+        // Get where clause of the current node check if condition row_num = 1 is present or not
+        if(selectQuery.getWhereClause() != null) {
+            ASTNodes.ASTExpression expressions_where = selectQuery.getWhereClause().getExpression();
+            if (expressions_where instanceof ASTNodes.ASTAndExpr) {
+                ImmutableList<ASTNodes.ASTExpression> list = ((ASTNodes.ASTAndExpr) expressions_where).getConjuncts();
+                for (ASTNodes.ASTExpression andExpr : list) {
+                    if (andExpr instanceof ASTNodes.ASTBinaryExpression) {
+                        Pair<String, Boolean> p = getWhereFilters(andExpr, false, whereClauseAlias, rowNumAlias);
+                        whereClauseAlias = p.getFirst();
+                        rhsOne = p.getSecond();
+                    }
+                    if (rhsOne)
+                        break;
+                }
+            } else {
+                Pair<String, Boolean> p = getWhereFilters(expressions_where, false, whereClauseAlias, rowNumAlias);
+                whereClauseAlias = p.getFirst();
+                rhsOne = p.getSecond();
+            }
+        }
+        if(rownumFlag && whereClauseAlias.equals(rowNumAlias) && rhsOne){
+            int lineNum = ZetaSQLStringParsingHelper.countLine(query, rowNumLineNum);
+            result.add(String.format(LATEST_RECORDS_ROW_NUMBER_PATTERN, lineNum));
+        }
+
+
+    }
+    //Traversing where clause to get lhs and rhs 
+    public Pair<String, Boolean> getWhereFilters(ASTNodes.ASTExpression expr, boolean rhsOne, String whereClauseAlias, String rowNumAlias){
```

Comment:
There are two extra things we should check to consider a WHERE filter as valid for the antipattern:
1. The operation of the `BinaryExpression` should be `=`
    * Otherwise, something like `row_number != 1` would trigger a false positive
    * Maybe we should also trigger the antipattern for `row_number > INT` and `row_number < INT`, those could also be implemented with `ARRAY_AGG()`
2. We're assuming `row_number` is always on the left and the literal `1` always on the right. It could be the other way around.